mlflow/pyfunc/__init__.py

```diff
@@ -716,6 +717,24 @@ def load_model(
     return PyFuncModel(model_meta=model_meta, model_impl=model_impl)
 
 
+def print_model_dependencies(model_uri):
+    """
+    Given a model URL, print model dependencies provided by model’s requirements.txt file.
+    And for databricks runtime, also print an instruction to update python environment with
+    the model dependencies.
+    """
+    local_path = _download_artifact_from_uri(artifact_uri=model_uri, output_path=None)
```

Comment:
Looks like we only need `requirements.txt`. Is it possible to only download it rather than the entire model artifacts?

Reply:
I have another idea, for each model URI, we can maintain a dict { model_uri --> downloaded_model_local_path } in memory, once we download a model to a temp local dir from model_uri, update the dict in memory. So that we can avoid download every time.

Reply:
> I have another idea, for each model URI, we can maintain a dict { model_uri --> downloaded_model_local_path } in memory, once we download a model to a temp local dir from model_uri, update the dict in memory. So that we can avoid download every time.

Do we really need this optimization? I don't think we do for now.

Reply:
I hope we can do this optimization. It is easy. And note some model might be large. This optimization is very meaningful on databricks. Because:

current code by default it downloaded into a temporary directory created by tempfile, the temp directory is not delete until OS reboot. For databricks users, their all notebook processes are running on driver machine, and OS does not reboot until cluster restart.

Some model might be very big., e.g. some deep-learning model, it might be several GBs size.

Reply:
I think it's best to download individual files, rather than the whole model. For pip requirements, we should download `requirements.txt`. For conda, we should download MLModel and use the `env` attribute of the pyfunc flavor spec to find the `conda.yaml` file & download it.